src/ec/curve25519/ops.rs

```diff
@@ -121,6 +121,79 @@ fn encode_point(x: Elem, y: Elem, z: Elem) -> EncodedPoint {
     bytes
 }
 
+// If the target architecture is non-Windows x86-64 or NEON-capable ARM, it is
+// faster to do scalar base point multiplication using the generic scalar
+// multiplication implementation.
+#[cfg(all(target_arch = "x86_64", not(windows)))]
+pub fn scalar_mult_with_base_point(scalar: &Scalar) -> EncodedPoint {
+    let mut point = [0u8; ELEM_LEN];
+    unsafe { GFp_x25519_scalar_mult(&mut point, scalar, &BASE_POINT); }
+    point
+}
+
+// If the target architecture is NEON-capable 32-bit ARM, then base point
+// scalar multiplication can be done faster using the architecture-optimized
+// generic scalar multiplication implementation. When the `target_feature`
+// compilation attribute is stabilized, we can remove the runtime check.
```

Comment:
Let's remove “When the `target_feature` compilation attribute is stabilized, we can remove the runtime check.” More likely, we'll add a feature that says "replace runtime CPU capability checks with compile-time CPU capability checks" and that will allow for the removal of the runtime check when that bit is flipped. But there are always going to be platforms (e.g. some 32-bit Android) where we will do a runtime check, indefinitely.